### PR TITLE
Fix GetNodes - Replace with streaming rpc

### PR DESCRIPTION
**Pull request by Joerger**
EDIT: https://github.com/gravitational/teleport/pull/7415 implements `ListNodes` with pagination and will most likely replace this PR.

Follow up PR to gravitational/teleport#7409 which removed the simple GetNodes rpc.

GetNodes should be either a streaming rpc or paginated rpc to properly handle clusters with a large number of nodes without exceeding the rpc return limit - gravitational/teleport#7402

### CODE REVIEW

**Comment by Joerger** [comment]
@russjones @r0mant Please review

**Comment by tcsc** [comment]
LGTM in general. What's the rationale for changing to streaming the nodes vs one big response?

File: lib/auth/grpcserver.go [hunk 1]
```diff
@@ -2445,6 +2446,29 @@ func (g *GRPCServer) GetNodes(ctx context.Context, req *types.ResourcesInNamespa
 	}, nil
 }
 
+// GetNodesStream returns a stream of all nodes in the given namespace.
+func (g *GRPCServer) GetNodesStream(req *types.ResourcesInNamespaceRequest, stream proto.AuthService_GetNodesStreamServer) error {
+	auth, err := g.authenticate(stream.Context())
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	nodes, err := auth.ServerWithRoles.GetNodes(stream.Context(), req.Namespace)
```
Comment (tcsc) [inline lib/auth/grpcserver.go:2455]: Do we care that the entire list of nodes is still being loaded into memory here, even though it is being streamed out piecemeal below?

**Review by tcsc — APPROVED** [review]

File: lib/auth/grpcserver.go (hunk 1 above)
Comment (Joerger) [reply lib/auth/grpcserver.go:2455]: For this bug fix we don't care, but that would definitely be a good optimization to follow up with. Would you just query the database with pagination to achieve that?

**Review by r0mant — COMMENTED** [review]
Fix lgtm, please add a test though.

Also, probably out of scope for this PR but could you also add TODO items to convert other List APIs to streaming as well (K8s clusters, apps, databases).

**Comment by klizhentas** [comment]
@Joerger curious, have you considered pagination here instead of stream? Usually streams are used for a continuously updated information, while pagination is better if you have to fetch the existing set of data. This use-case seems to be closer to the second one. Having pagination has another advantage - you can fetch chunks of nodes in parallel.

**Comment by Joerger** [comment]
Good point, I'll update this PR to use pagination and I made a new issue for updating other rpcs - https://github.com/gravitational/teleport/issues/7412. I don't think our backend currently supports fetching chunks in parallel though.

**Comment by Joerger** [comment]
Replaced with pagination implementation - https://github.com/gravitational/teleport/pull/7415

**Pull request closed without merging**